src/subscriptions/subscriptionSelectors.js

```diff
@@ -27,6 +27,11 @@ export const getSubscriptionsById: Selector<Map<number, Subscription>> = createS
     new Map(subscriptions.map(subscription => [subscription.stream_id, subscription])),
 );
 
+export const getSubscriptionsByName: Selector<Map<string, Subscription>> = createSelector(
```

Comment:
Is `name` (the stream name) a unique identifier for `Subscription`s? In subscriptionsReducer, it looks like `stream_id` is considered unique:

```
    case EVENT_SUBSCRIPTION:
      switch (action.op) {
        case 'add':
          return state.concat(
            action.subscriptions.filter(x => !state.find(y => x.stream_id === y.stream_id)),
          );
```

Perhaps they're both unique? The existence of the [`get_stream_id` endpoint](https://zulipchat.com/api/get-stream-id) suggests so. Maybe it would be useful to have code comments on certain API types about any unique identifiers they have.

Reply:
Yeah, they are. It'd be better to use the numeric ID, but that's a long-open ongoing migration. I just filed #3918 to describe it.

The one real disadvantage of stream names vs. IDs -- other than the disadvantages of having *two* different ways, which complicates the code -- is that names can change.

The reason this codepath is doing its lookup by name is ultimately that it has a `Narrow`, and our `Narrow` type uses emails for users (cf #3764) and names for streams. We should fix that, but it'll be rather a project in itself.